pkg/registry/apis/provisioning/read.go

```diff
@@ -55,19 +57,53 @@ func (s *readConnector) Connect(ctx context.Context, name string, opts runtime.O
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		idx := strings.Index(r.URL.Path, "/"+name+"/read")
-		filePath := strings.TrimLeft(r.URL.Path[idx+len(name+"/read")+2:], "/")
+		filePath := strings.TrimLeft(r.URL.Path[idx+len(name+"/read")+1:], "/")
 		if filePath == "" {
 			responder.Error(errors.NewBadRequest("missing path"))
 			return
 		}
 		commit := r.URL.Query().Get("commit")
 
-		rsp, err := repo.ReadResource(ctx, filePath, commit)
+		reader, err := repo.ReadResource(ctx, filePath, commit)
 		if err != nil {
 			responder.Error(err)
-		} else {
-			responder.Object(200, rsp)
+			return
 		}
+
+		obj, _, err := LoadYAMLOrJSON(reader)
+		if err != nil {
+			// Reset the buffer and try reading a dashboard
+			var data []byte
+			rr, ok := reader.(io.Seeker)
+			if ok {
+				_, err = rr.Seek(0, io.SeekStart)
+			}
+			if err != nil {
+				// Try reading again
+				reader, err = repo.ReadResource(ctx, filePath, commit)
+				if err != nil {
+					responder.Error(err)
+					return
+				}
+			}
+			data, err = io.ReadAll(reader)
+			if err != nil {
+				responder.Error(err)
+				return
+			}
```

Comment:
^^^ UGLY... any better ideas?

We could always save the raw bytes 🤷🏻  -- or maybe require the initial reader is also a seeker?

Reply:
I wouldn't mind just passing around `[]byte`. I can't imagine anyone having a gig of dashboard JSON anyways.

Reply:
ya, likely not worth trying to stream.  I'll update